File: csv-machine-problems/JAVA/MP11_FrequencyCount.java
// MP11 - Frequency Count for Column Values
// Student: Gray Allen Arizala

import java.io.*;
import java.util.*;

public class MP11_FrequencyCount {

    public static void main(String[] args) {

        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter CSV dataset file path: ");
        String filePath = scanner.nextLine();

        System.out.print("Enter column index to analyze: ");
        int columnIndex = scanner.nextInt();

        HashMap<String, Integer> frequency = new HashMap<>();

        try {

            BufferedReader br = new BufferedReader(new FileReader(filePath));
            String line;

            while ((line = br.readLine()) != null) {

                String[] columns = line.split(",");

                if (columnIndex < columns.length) {

                    String value = columns[columnIndex];

                    frequency.put(value, frequency.getOrDefault(value, 0) + 1);
                }
            }

            br.close();

            System.out.println("\nFrequency Count:");

            for (String key : frequency.keySet()) {
                System.out.println(key + " : " + frequency.get(key));
            }

        } catch (IOException e) {
            System.out.println("Error reading dataset.");
        }
    }
}

File: Midtermlab3/mp11_frequencyCount.js
// MP11 - Frequency Count for Column Values
// Student: Gray Allen Arizala

const fs = require('fs');
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.question("Enter CSV dataset file path: ", function(filePath) {

    rl.question("Enter column index: ", function(columnIndex) {

        try {

            const data = fs.readFileSync(filePath, 'utf8');
            const rows = data.split('\n');

            const frequency = {};

            rows.forEach(row => {

                const columns = row.split(",");

                if (columnIndex < columns.length) {

                    const value = columns[columnIndex];

                    frequency[value] = (frequency[value] || 0) + 1;

                }

            });

            console.log("\nFrequency Count:");

            for (const key in frequency) {
                console.log(key + " : " + frequency[key]);
            }

        } catch (error) {
            console.log("Error reading dataset.");
        }

        rl.close();

    });

});